apps/synchronize/src/github/eventHerpers.test.js

```diff
@@ -0,0 +1,230 @@
+        organizationId: organization.id,
+      });
+      const account = await getAccountOrThrow(
+        ORGANIZATION_PURCHASE_EVENT_PAYLOAD
+      );
+      expect(account).toMatchObject({
+        userId: null,
+        organizationId: organization.id,
+      });
+    });
+  });
+
+  describe("#getOrCreateAccount", () => {
+    let account;
+
+    it("should create new user and account", async () => {
+      const usersCount = await User.query().resultSize();
+      const accountsCount = await Account.query().resultSize();
+      const account = await getOrCreateAccount(USER_PURCHASE_EVENT_PAYLOAD);
+      const createdUser = await account.$relatedQuery("user");
+      expect(createdUser.githubId).toBe(
+        USER_PURCHASE_EVENT_PAYLOAD.marketplace_purchase.account.id
+      );
+
+      const newUsersCount = await User.query().resultSize();
+      expect(newUsersCount).toBe(usersCount + 1);
+
+      const newAccountsCount = await Account.query().resultSize();
+      expect(newAccountsCount).toBe(accountsCount + 1);
+    });
+
+    it("should create new organization and account", async () => {
+      const organizationsCount = await Organization.query().resultSize();
+      const accountsCount = await Account.query().resultSize();
+      const account = await getOrCreateAccount(
+        ORGANIZATION_PURCHASE_EVENT_PAYLOAD
+      );
+      const createdOrganization = await account.$relatedQuery("organization");
+      expect(createdOrganization.githubId).toBe(
+        ORGANIZATION_PURCHASE_EVENT_PAYLOAD.marketplace_purchase.account.id
+      );
+
+      const newOrganizationsCount = await Organization.query().resultSize();
+      expect(newOrganizationsCount).toBe(organizationsCount + 1);
+
+      const newAccountsCount = await Account.query().resultSize();
+      expect(newAccountsCount).toBe(accountsCount + 1);
+    });
+
+    it("should return existing user account", async () => {
+      const { id: githubId } =
+        USER_PURCHASE_EVENT_PAYLOAD.marketplace_purchase.account;
+      const user = await factory.create("User", { githubId });
+      await factory.create("UserAccount", { userId: user.id });
+      const account = await getOrCreateAccount(USER_PURCHASE_EVENT_PAYLOAD);
+      expect(account.userId).toBe(user.id);
+    });
+
+    it("should return existing organization account", async () => {
+      const { id: githubId } =
+        ORGANIZATION_PURCHASE_EVENT_PAYLOAD.marketplace_purchase.account;
+      const organization = await factory.create("Organization", { githubId });
+      await factory.create("OrganizationAccount", {
+        organizationId: organization.id,
+      });
+      account = await getOrCreateAccount(ORGANIZATION_PURCHASE_EVENT_PAYLOAD);
+      expect(account).toMatchObject({ organizationId: organization.id });
+    });
+  });
+
+  describe("#getActivePurchaseOrThrow", () => {
+    it("should throw when account is not provided", async () => {
+      expect.assertions(1);
```

Comment:
You can also do:
```js
await expect(getActivePurchaseOrThrow()).rejects.toMatch('can't find purchase of missing account')
```

Reply:
Since I never remember the syntax, I use a simple try / catch + `expect.assertions`. But it is tricky because if you remove the first line your test become irrelevant and the two lines are not linked and not statically checkable.

Your version is way better @salper.